app/assets/javascripts/modals.js

```diff
@@ -1,44 +1,88 @@
-var minimumWidthToDisplayModal = 768;
-var modal;
-
-function loadModal(modalTarget) {
-  console.log(viewportWidth());
-  if(viewportWidth() < minimumWidthToDisplayModal) {
-    return function(evt) { console.debug("Modal not displayed because minimum width not met") }
-  }
-
-  return function(evt) {
-    evt.preventDefault();
-    return $("#modal").load(modalTarget).dialog({
-      position: { my: "center top", at: "center top", of: evt.target },
-      modal: true,
-      draggable: false,
-      resizeable: false,
-      width: 500,
-      closeText: "Not now",
-      title: ''
-    });
-  }
-}
-function closeModal() { modal.dialog('close') }
-
-function modalActivation() {
-  $('a.sign-up').on('click', function(evt) {
-    if(evt.currentTarget.href == '/signup') {
-      modal = loadModal('/signup')(evt)
+$(function() {
+  'use strict';
+  var modal,
+    viewportWidth,
+    loadModal,
+    closeModal,
+    modalActivation;
+
+  /**
+   * get viewport width
+   * @return {int} width of the viewport
+   */
+  viewportWidth = function() {
+    return (window.innerWidth > 0) ? window.innerWidth : screen.width;
+  };
+
+  /**
+   * Load modals with given modalTarget
+   * @param  {string} modalTarget modal target element selector
+   * @return {callback}             function that generates modal
+   */
+  loadModal = function(modalTarget) {
+    var minimumWidthToDisplayModal = 768;
+    /**
+     * Generate a modal window from jQuery
+     * @param  {object} evt browser event object
+     * @return {callback}     function that loads a modal from the modal target
+     */
+    var generateModal = function(evt) {
+      evt.preventDefault();
+
+      return $('#modal').load(modalTarget).dialog({
+        position: {
+          my: 'center top',
+          at: 'center top',
+          of: evt.target
+        },
+        modal: true,
+        draggable: false,
+        resizeable: false,
+        width: 500,
+        closeText: 'Not now',
+        title: ''
+      });
+    };
+
+    // console.log(viewportWidth());
+    if (viewportWidth() < minimumWidthToDisplayModal) {
+      return function() {
+        // console.debug("Modal not displayed because minimum width not met")
+      };
     }
-  });
 
-  $('.tea-time-scheduling').on('click', function(evt) {
-    modal = loadModal(evt.currentTarget.href)(evt)
-  });
+    return generateModal;
+  };
 
-  $('.cities.show a.cancel').on('click', function(evt) {
-    console.log(evt)
-    evt.preventDefault();
-    closeModal();
-  })
-}
+  /**
+   * Close modal window
+   */
+  closeModal = function() {
+    modal.dialog('close');
+  };
+
+  /**
+   * Activate Modal Callbacks
+   */
+  modalActivation = function() {
+    // $('.sign-up-emphasis').on('click', function(evt) {
+    //   if ($(evt.currentTarget).attr('href') === '/signup') {
+    //     modal = loadModal('/signup')(evt);
+    //   }
+    // });
```

Comment:
let's not comment out code. let's delete it entirely if it's meant to be deleted.